packetbeat/statsdns/statistics_dns.go

```diff
@@ -101,62 +101,80 @@ var (
 	IpsServer                    []string
 	UrlAnnouncementDeployFromBam string
 	MapViewIPs                   map[int]map[string][]string
+	QStatDNS                     *QueueStatDNS
+	IsActive                     bool
+	SubActive                    bool
 )
 
 func InitStatisticsDNS() {
 	GetConfigDNSStatistics()
+	// Create chan for management Statistic DNS counter
+	QStatDNS = NewQueueStatDNS()
+	IsActive := true
```

Comment:
Are you shadowing the variables? Seems you are trying to use the local var instead of the global one. This means you cannot stop the loop by calling `Stop` function.